docs/integrations/jupiter.md

```diff
@@ -0,0 +1,438 @@
+
+@jup-ag/core is the Core package used to interact with jupiter on-chain programs to perform swaps between two possible token pairs.
+
+<CodeGroup>
+  <CodeGroupItem title="YARN" active>
+
+```bash
+yarn add @jup-ag/core
+```
+
+  </CodeGroupItem>
+
+  <CodeGroupItem title="NPM">
+
+```bash
+npm install @jup-ag/core
+```
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Fetching Token list from Jupiter
+
+All the possible tokens that can be swapped with jupiter for a given network is being fetched.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/token-list/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/token-list/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Loading the Jupiter instance
+
+Jupiter instace is being created with the provided configurations. There are alot of options present to tinker with to know more about it. go [here](https://docs.jup.ag/jupiter-core/full-guide)
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/loading-instance/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/loading-instance/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Getting the RouteMap
+
+The RouteMap identifies what tokens you can swap to given an input token. The route map only contains token mint addresses and no meta data.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/route-map/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/route-map/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Getting the routes for given Input and Output token
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/routes/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/routes/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Execute the Token Swap
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/routes/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/routes/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+## How to use Jupiter in a Client Application
+
+### Installation
+
+<CodeGroup>
+  <CodeGroupItem title="YARN" active>
+
+```bash
+yarn add @jup-ag/react-hook
+```
+
+  </CodeGroupItem>
+
+  <CodeGroupItem title="NPM">
+
+```bash
+npm install @jup-ag/react-hook
+```
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Adding the Provider
+
+We need to setup the JupiterProvider in order to use the useJupiter Hook Through the React App.
```

Comment:
Why suddenly jumping into React?

Reply:
Hi, I'm Dickson from Jupiter,
Jupiter have different library that caters to the different needs of the client (web, mobile, native), therefore @0xPratik included different section/method of interacting with Jupiter.

There's mainly three way you can interact with Jupiter right now: 
1. Core
2. React hook
3. API 

Let us know if that clears up a little.